diff --git a/build/hooks/upload-sourcemaps.js b/build/hooks/upload-sourcemaps.js
index 0ed5c213d2e65a30fe8d8d13ef319f2b2962511e..d7f5591294e6b76f6364a4ae874e99f4a0dc03a1 100644
--- a/build/hooks/upload-sourcemaps.js
+++ b/build/hooks/upload-sourcemaps.js
@@ -16,10 +16,14 @@ module.exports = async (options) => {
     mkdirp_1.default.sync(tmpdir);
     try {
         // We use the same filenames for sourcemaps as Sentry does (even though the naming is unfortunate)
-        fs_1.default.writeFileSync(tmpdir + '/main.jsbundle', iosBundle, 'utf-8');
-        fs_1.default.writeFileSync(tmpdir + '/main.jsbundle.map', iosSourceMap, 'utf-8');
-        fs_1.default.writeFileSync(tmpdir + '/index.android.bundle', androidBundle, 'utf-8');
-        fs_1.default.writeFileSync(tmpdir + '/index.android.bundle.map', androidSourceMap, 'utf-8');
+        if (!config?.platform || config.platform === 'ios') {
+          fs_1.default.writeFileSync(tmpdir + '/main.jsbundle', iosBundle, 'utf-8');
+          fs_1.default.writeFileSync(tmpdir + '/main.jsbundle.map', iosSourceMap, 'utf-8');
+        }
+        if (!config?.platform || config.platform === 'android') {
+          fs_1.default.writeFileSync(tmpdir + '/index.android.bundle', androidBundle, 'utf-8');
+          fs_1.default.writeFileSync(tmpdir + '/index.android.bundle.map', androidSourceMap, 'utf-8');
+        }
         if (!config) {
             log('No config found in app.json, falling back to environment variables...');
         }
